### ISSUE TITLE
Can you add support to sort a pivot table on BelongsToMany.

### ISSUE THREAD

**Issue by 949mac** [optimistdigital/nova-sortable#7]
![image](https://user-images.githubusercontent.com/23151962/73985515-395d7b00-48f0-11ea-96d2-b5c39dc9784a.png)

It's sorting the primary model rather  than the pivot.

**Comment by Tarpsvo** [comment]
Hey! Sorry for the late reply. Adding support for pivot tables is not that easy, I need to give it some thought.

Meanwhile, the newly released version 1.2.0 no longer shows the sort buttons on pivot tables, due to the simple reason that it didn't work.

Thanks!

**Comment by sicbe** [comment]
Might it be possible if we add the search query to the model?

` public function buildSortQuery()
    {
        return static::query()->where('user_id', $this->user_id);
    }`

https://github.com/spatie/eloquent-sortable#grouping

**Comment by Tarpsvo** [comment]
> Might it be possible if we add the search query to the model?
> 
> ` public function buildSortQuery() { return static::query()->where('user_id', $this->user_id); }`
> 
> https://github.com/spatie/eloquent-sortable#grouping

Hmm, can you explain what's stopping you from adding it to your model right now?

**Comment by sicbe** [comment]
It was as a suggestion to the original poster :-). I was trying to the same thing earlier this week so found that in the docs of the eloquent-sortable.

I tried it yesterday and finally got it to work to sort on the pivot table.

**Comment by Tarpsvo** [comment]
Ah, that makes sense!

I should probably document this and re-enable sorting on pivot lists.

**Comment by 949mac** [comment]
Thanks for the responses!

**Comment by Tarpsvo** [comment]
Sorting pivot tables is now supported in version 1.3.0. Instructions in README. Let me know if it works for you, good luck!

**Comment by potibas** [comment]
Is there an example that could be used for reference? I'm having a bit of difficulty to implement this feature just from the instructions on the README file, keep getting errors on nova's resource pages (sort_order not found).

**Comment by phlisg** [comment]
@potibas 

I managed to make it work on my side, the provided example was a bit confusing at first, but I figured it out. If it helps you, here's my implementation on my side:

#### TL; DR: just add on your "main model" a column `sort_order`

## Models:
`Visits` (Guided tours if you will)
`Stages` (guided tours "steps" if you will)
- has a column named `order` (for manual ordering, if I couldn't find a package to order relations)

### Pivot Model
`VisitStages` (using table `visit_stages`)
- has a column `sort_order`

> So schematically, the relations look like this:
> `Visits` **>** `VisitStages` **<** `Stages`

On `Stage`:
```php
class Stage extends Model implements Sortable 
{ 
    use SortableTrait;

    public $sortable = [
      'order_column_name' => 'order', // here using the "deprecated" column, otherwise I get the error 'sort_order not found'
      `sort_when_creating` => true,
      `sort_on_pivot` => true, // this is mis-leading
    ];

   public function visits()
   {  
      return $this->belongsToMany(Visit::class, 'visit_stages')
         ->withPivot('sort_order') // your pivot order column
         ->using(VisitStages::class)
         ->orderBy('visit_stages.sort_order');
   }
}
```

On `Visit`:
```php
// ...
public function stages()
{
    return $this->belongsToMany(...); // same as Stage::visits()
}
// ...
```

On `VisitStages`:
```php
class VisitStages extends Pivot
{
    protected $table = 'visit_stages';

    public static $sortable = [
        'sort_order'
    ];
}
```

And finally, the `Stage` resource:
```php
class Stage extends Resource
{
    use HasSortableRows;
    // ...
}
